go-controller/pkg/clustermanager/userdefinednetwork/template/net-attach-def-template_test.go

```diff
@@ -218,39 +212,57 @@ var _ = Describe("NetAttachDefTemplate", func() {
+			"topology Layer2 is specified but Layer2 config is nil",
 		),
 		Entry("CUDN, invalid topology: topology layer2 & localnet config",
 			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
 				Topology: udnv1.NetworkTopologyLayer2, Localnet: &udnv1.LocalnetConfig{}}}},
+			"topology Layer2 is specified but Layer2 config is nil",
 		),
 		Entry("CUDN, invalid topology: topology layer3 & layer2 config",
 			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
 				Topology: udnv1.NetworkTopologyLayer3, Layer2: &udnv1.Layer2Config{}}}},
+			"topology Layer3 is specified but Layer3 config is nil",
 		),
 		Entry("CUDN, invalid topology: topology layer3 & localnet config",
 			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
 				Topology: udnv1.NetworkTopologyLayer3, Localnet: &udnv1.LocalnetConfig{}}}},
+			"topology Layer3 is specified but Layer3 config is nil",
 		),
 		Entry("CUDN, invalid topology: topology localnet & layer2 config",
 			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
 				Topology: udnv1.NetworkTopologyLocalnet, Layer2: &udnv1.Layer2Config{}}}},
+			"topology Localnet is specified but Localnet config is nil",
 		),
 		Entry("CUDN, invalid topology: topology localnet & layer3 config",
 			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
 				Topology: udnv1.NetworkTopologyLocalnet, Layer3: &udnv1.Layer3Config{}}}},
+			"topology Localnet is specified but Localnet config is nil",
+		),
+		Entry("CUDN, localnet: IPv4 excludeSubnets not in range of subnets",
+			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
+				Topology: udnv1.NetworkTopologyLocalnet,
+				Localnet: &udnv1.LocalnetConfig{Role: udnv1.NetworkRoleSecondary, PhysicalNetworkName: "localnet1",
+					Subnets:        udnv1.DualStackCIDRs{"192.168.0.0/16", "2001:dbb::/64"},
+					ExcludeSubnets: []udnv1.CIDR{"192.200.0.0/30"},
+				},
+			}}},
+			"do not contain exluded subnets 192.200.0.0/30",
 		),
-		Entry("CUDN, localnet: excludeSubnets not in range of subnets",
+		Entry("CUDN, localnet: IPv6 excludeSubnets not in range of subnets",
 			&udnv1.ClusterUserDefinedNetwork{Spec: udnv1.ClusterUserDefinedNetworkSpec{Network: udnv1.NetworkSpec{
 				Topology: udnv1.NetworkTopologyLocalnet,
 				Localnet: &udnv1.LocalnetConfig{Role: udnv1.NetworkRoleSecondary, PhysicalNetworkName: "localnet1",
 					Subnets:        udnv1.DualStackCIDRs{"192.168.0.0/16", "2001:dbb::/64"},
-					ExcludeSubnets: []udnv1.CIDR{"192.200.0.0/30", "2001:aaa::/127", "192.300.0.1/32", "2001:bbb::1/120"},
+					ExcludeSubnets: []udnv1.CIDR{"2001:aaa::/127"},
 				},
 			}}},
+			"do not contain exluded subnets 2001:aaa::/127",
```

Comment:
note: "exluded" is spelled wrong in the error message. Want to fix that?

Reply:
fixed